library/Fission/User/Mutate.hs

```diff
@@ -22,13 +26,28 @@ import qualified Fission.User.Table as Table
 
 -- | Create a new, timestamped entry
 create :: MonadRIO    cfg m
+       => MonadSelda      m
+       => HasLogFunc cfg
+       => SecretDigest
+       -> m (ID User)
+create sekret = do
+  now <- liftIO getCurrentTime
+
+  uID <- insertWithPK Table.users
+    [User def Regular True Nothing sekret <@ now]
```

Comment:
Knowledge share: could you explain to me what `def` does here?

Reply:
Not completely sure on this one tbh, but here's my understanding:
`def` is short for [default](https://hackage.haskell.org/package/selda-0.5.0.0/docs/Database-Selda.html#v:def)

Check the documentation for [Database.Selda.insert](https://hackage.haskell.org/package/selda-0.5.0.0/docs/Database-Selda.html#v:insert)
Here, `def` is passed in in place of `userID`. We don't want to pass in an actual userID (since it's auto-incrementing). So `def` tells Selda to default `userID`. In this case to `autoPrimary` (see `Fission.User.Table` for table definition)

Reply:
@dholms is correct 👍

Reply:
Awesome thanks @dholms !!